### PR TITLE
[FIX] Collapse ITK transforms in only one file

### LINKED ISSUE #623: Number of files written to working directory
Running fmriprep on MyConnectome generated at least 1,794,910 files in the working directory.  My job stopped because I exceeded the maximum number of files on my system (sherlock:oak - 8 mil).  Something to keep in mind :-)

### PR DESCRIPTION
**Pull request by oesteban**
Fixes #623

### CODE REVIEW

File: fmriprep/workflows/bold.py [hunk 1]
```diff
@@ -683,9 +684,7 @@ def init_bold_hmc_wf(metadata, bold_file_size_gb, ignore,
     hmc = pe.Node(fsl.MCFLIRT(save_mats=True, save_plots=True),
                   name='BOLD_hmc', mem_gb=bold_file_size_gb * 3)
 
-    hcm2itk = pe.MapNode(c3.C3dAffineTool(fsl2ras=True, itk_transform=True),
-                         iterfield=['transform_file'], name='hcm2itk',
-                         mem_gb=0.05)
+    hcm2itk = pe.Node(MCFLIRT2ITK(), name='hcm2itk', mem_gb=0.05)
```
Comment (oesteban) [inline fmriprep/workflows/bold.py]: @effigies what would you say it is best: setting `n_procs` here to `omp_nthreads` or leaving it serial? Locally I see the linear improvement with parallelism and I'd expect large improvements for very long BOLD files. WDYT?

Comment (effigies) [reply fmriprep/workflows/bold.py]: Yeah, parallelizing sounds good. (FYI I haven't had a chance to review this in detail, yet.)

File: fmriprep/workflows/bold.py [hunk 2]
```diff
@@ -683,9 +687,7 @@ def init_bold_hmc_wf(metadata, bold_file_size_gb, ignore,
     hmc = pe.Node(fsl.MCFLIRT(save_mats=True, save_plots=True),
                   name='BOLD_hmc', mem_gb=bold_file_size_gb * 3)
 
-    hcm2itk = pe.MapNode(c3.C3dAffineTool(fsl2ras=True, itk_transform=True),
-                         iterfield=['transform_file'], name='hcm2itk',
-                         mem_gb=0.05)
+    hcm2itk = pe.Node(MCFLIRT2ITK(), name='hcm2itk', mem_gb=0.05, n_procs=omp_nthreads)
```
Comment (effigies) [inline fmriprep/workflows/bold.py]: `MCFLIRT2ITK(nprocs=omp_nthreads)`

Comment (oesteban) [reply fmriprep/workflows/bold.py]: thanks!

File: fmriprep/interfaces/itk.py [hunk 3]
```diff
@@ -8,15 +8,166 @@
...
+    input_spec = MCFLIRT2ITKInputSpec
+    output_spec = MCFLIRT2ITKOutputSpec
+
+    def _run_interface(self, runtime):
+        tmp_folder = TemporaryDirectory()
+        parallel = Parallel(n_jobs=self.inputs.nprocs)
+        itk_outs = parallel(delayed(_mat2itk)(
+            in_mat, self.inputs.in_reference, self.inputs.in_source, i,
+            tmp_folder.name) for i, in_mat in enumerate(self.inputs.in_files))
+        tmp_folder.cleanup()
```
Comment (effigies) [inline fmriprep/interfaces/itk.py]: It's fine how it is, but I find it a little cleaner to use `TemporaryDirectory()` as a context manager:

```Python
parallel = Parallel(n_jobs=self.inputs.nprocs)
with TemporaryDirectory() as tmp_folder:
    itk_outs = parallel(delayed(_mat2itk)(
        in_mat, self.inputs.in_reference, self.inputs.in_source, i,
        tmp_folder.name) for i, in_mat in enumerate(self.inputs.in_files))
```

File: fmriprep/interfaces/itk.py [hunk 4]
```diff
@@ -8,15 +8,166 @@
...
+        self._results['out_file'] = os.path.abspath('mat2itk.txt')
+        with open(self._results['out_file'], 'w') as f:
+            f.write(tfms)
+
+        return runtime
+
+
+class MultiApplyTransformsInputSpec(ApplyTransformsInputSpec):
+    input_image = InputMultiPath(File(exists=True), mandatory=True,
+                                 desc='list of MAT files from MCFLIRT')
```
Comment (effigies) [inline fmriprep/interfaces/itk.py]: I think this desc should be a 'list of input images'?

**Comment by effigies** [comment]
I'll merge this, but thinking about it, we use the HMC workflow's `bold_split` output in two places, both of which feed into the new `MultiApplyTransforms`. One option to again reduce the number of files would be to run `fsl.Split` as an interface inside `MultiApplyTransforms` and place the outputs in the temporary directory, as well. That would be conceptually a little cleaner, too, though it does duplicate work.

**Pull request merged**